packages/core/test/reducers/core.test.ts

```diff
@@ -562,6 +562,44 @@ test('core reducer - update - should update errors', (t) => {
   });
 });
 
+test('core reducer - update - setting a state slice as undefined should remove the slice', (t) => {
+  const schema = {
+    type: 'object',
+    properties: {
+      foo: {
+        type: 'string',
+      },
+      fizz: {
+        type: 'string',
+      },
+    },
+  };
+
+  const before: JsonFormsCore = {
+    data: {
+      foo: 'bar',
+      fizz: 42,
+    },
+    schema: schema,
+    uischema: {
+      type: 'Label',
+    },
+    errors: [],
+    validator: new Ajv().compile(schema),
+  };
+
+  const after = coreReducer(
+    before,
+    update('foo', (_) => {
+      return undefined;
+    })
+  );
+
+  t.not(before, after);
+  t.not(before.data, after.data);
+  t.like(Object.keys(after.data), ['fizz']);
```

Comment:
`t.like` must not be used. It leads to an out of memory exception of NodeJS. Interestingly enough, in VSCode the `like` method is not even recognized at all. Why this leads to out of memory and not just fails the test execution, I don't know.
To fix this, you can use `t.deepEqual` to verify the `after` object has the correct keys. See the suggested fix below. 
```suggestion
  t.deepEqual(Object.keys(after.data), ['fizz']);
```

Reply:
Thanks for your input, included in [041db92](https://github.com/eclipsesource/jsonforms/pull/2163/commits/041db924a68ea824abe81aa265e5b2fe3a97869b)

Reply:
All green, thanks!